src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -309,50 +350,11 @@ private bool StartCore(ProcessStartInfo startInfo)
                 {
                     throw new Win32Exception(SR.DirectoryNotValidAsInput);
                 }
-            }
-
-            if (string.IsNullOrEmpty(filename))
-            {
-                throw new Win32Exception(Interop.Error.ENOENT.Info().RawErrno);
-            }
-
-            bool setCredentials = !string.IsNullOrEmpty(startInfo.UserName);
-            uint userId = 0;
-            uint groupId = 0;
-            if (setCredentials)
-            {
-                (userId, groupId) = GetUserAndGroupIds(startInfo);
-            }
 
-            // Lock to avoid races with OnSigChild
-            // By using a ReaderWriterLock we allow multiple processes to start concurrently.
-            s_processStartLock.EnterReadLock();
-            try
-            {
-                // Invoke the shim fork/execve routine.  It will create pipes for all requested
-                // redirects, fork a child process, map the pipe ends onto the appropriate stdin/stdout/stderr
-                // descriptors, and execve to execute the requested process.  The shim implementation
-                // is used to fork/execve as executing managed code in a forked process is not safe (only
-                // the calling thread will transfer, thread IDs aren't stable across the fork, etc.)
-                Interop.Sys.ForkAndExecProcess(
-                    filename, argv, envp, cwd,
+                ForkAndExecProcess(filename, argv, envp, cwd,
```

Comment:
also - I believe we should throw if this returns false

Reply:
This never returns false, it throws.